packages/ui/core-components/src/lib/atoms/inputs/date-range/DateRange.svelte

```diff
@@ -52,30 +52,46 @@
 	}
 
 	const YYYYMMDD = /^\d{4}-\d{2}-\d{2}$/;
-	$: startString =
-		typeof start === 'string' && YYYYMMDD.test(start)
-			? start
-			: start instanceof Date
-				? dateToYYYYMMDD(start)
-				: $query?.[0].start instanceof Date
-					? dateToYYYYMMDD($query?.[0].start)
-					: dateToYYYYMMDD(new Date(0));
-	$: endString =
-		typeof end === 'string' && YYYYMMDD.test(end)
-			? end
-			: end instanceof Date
-				? dateToYYYYMMDD(end)
-				: $query?.[0].end instanceof Date
-					? dateToYYYYMMDD($query?.[0].end)
-					: dateToYYYYMMDD(new Date());
+
+	let startString;
+	$: if (
+		typeof startString !== 'string' ||
+		!YYYYMMDD.test(startString) ||
+		startString === dateToYYYYMMDD(new Date(0))
+	) {
+		startString =
+			typeof start === 'string' && YYYYMMDD.test(start)
+				? start
+				: start instanceof Date
+					? dateToYYYYMMDD(start)
+					: $query?.[0].start instanceof Date
+						? dateToYYYYMMDD($query?.[0].start)
+						: dateToYYYYMMDD(new Date(0));
+	}
+
+	let endString;
+	$: if (
+		typeof endString !== 'string' ||
+		!YYYYMMDD.test(endString) ||
+		endString === dateToYYYYMMDD(new Date())
+	) {
+		endString =
+			typeof end === 'string' && YYYYMMDD.test(end)
+				? end
+				: end instanceof Date
+					? dateToYYYYMMDD(end)
+					: $query?.[0].end instanceof Date
+						? dateToYYYYMMDD($query?.[0].end)
+						: dateToYYYYMMDD(new Date());
+	}
 
 	$: $inputs[name] = { start: startString, end: endString };
 
 	let selectedDateRange;
 	$: if (selectedDateRange && (selectedDateRange.start || selectedDateRange.end)) {
 		$inputs[name] = {
-			start: dateToYYYYMMDD(selectedDateRange.start?.toDate(getLocalTimeZone()) ?? new Date(0)),
-			end: dateToYYYYMMDD(selectedDateRange.end?.toDate(getLocalTimeZone()) ?? new Date())
+			start: selectedDateRange.start?.toString() ?? dateToYYYYMMDD(new Date(0)),
+			end: selectedDateRange.end?.toString() ?? dateToYYYYMMDD(new Date())
```

Comment:
Does this change cause a loss of timezone information?

Reply:
@ItsMeBrianD 
Actually 'selectedDateRange' object has start and end fields, both of which are `CalenderDate` objects.
As the user is directly selecting a date, timezone is not required. We need just the selected date. Converting to `Date` object can lead to internationalization issues. This is recommended in their docs:
https://react-spectrum.adobe.com/internationalized/date/CalendarDate.html#conversion
![image](https://github.com/evidence-dev/evidence/assets/54364088/37edbf60-5b4f-41e7-91f2-2c3e2147ba2f)


Luckily the `toString` method give the date string directly in required format. We can just skip the conversion to Date object and then to string in `dateToYYYYMMDD` fn.